inference-contract/docs/notebooks/templates/token.py

```diff
@@ -0,0 +1,182 @@
+#
+# The contract context defines the configuration for a collection of contract
+# objects that interact with one another. By default, the context file used in
+# this notebook is specific to the toke class used to mint the token.
+# We need the class to ensure that all of the information necessary for the token
+# itself is available. If you prefer to use a common context file, edit the
+# context_file variable below.
+#
+# For the most part, no other modifications should be required.
+
+# %%
+token_class_path = 'token.' + token_class
+context_file = bindings.expand(context_file)
+print("using context file {}".format(context_file))
+
+context = pc_jupyter.ml_jupyter.initialize_token_context(state, bindings, context_file, token_class_path)
+print('context initialized')
+
+# %% [markdown]
+# <hr style="border:2px solid gray">
+#
+# ## Operate on the Contract
+
+# %%
+token_context = pc_jupyter.pbuilder.Context(state, token_path)
+
+
+# %% [markdown]
+# ### Invoke Inference Operation
+#
+# The model is an image classfication model. Use the following cell to
+# provide input to the classification oepration. Current token object policy
+# simply checks ownership of the token. There are no additional policy checks
+# in this example.
+#
+# For the parameters to be provided below, if you are running the jupyter server
+# via the contracts docker container, copy your test image to
+# `$PDO_CONTRACTS_SRC_ROOT/docker/xfer/client/` folder and
+# provide the value `/project/pdo/xfer/client` for the input `path`.
+#
+# Also, please note that `$PDO_CONTRACTS_SRC_ROOT/inference-contract/data` folder
+# contains the sample image `zebra_wiki.jpg` that may be used for testing
+
+# %% tags=["parameters"]
+image_name = input('Name of the image file')
+path = input('Path on disk where the image is located')
+
+
+# %%
+inference_result = pc_jupyter.pcommand.invoke_contract_cmd(
+    pc_jupyter.ml_inference_token_object.cmd_do_inference,
+    state, token_context, image=image_name, search_path=[path])
+
+
+# %% [markdown]
+# ### Transfer Ownership to a New User
+#
+# User1 now transfers ownership of the token object (and hence its capabilities)
+# to a new user. Note that the notebook assumes that user1 knows the public key
+# of the new user, and is located at  `${keys}/`.
+
+# %% tags=["parameters"]
+new_user_identity = input('Name of the new User')
+
+
+# %%
+pc_jupyter.pcommand.invoke_contract_cmd(
+    pc_jupyter.ml_inference_token_object.cmd_transfer_assets,
+    state, token_context, new_owner=new_user_identity)
+
+# %% [markdown]
+# ### Test Ownership
+#
+# Let's ensure that user1 can no longer perform operations on the asset.
+# The following asset usage command should fail:
+
+# %%
+inference_result = pc_jupyter.pcommand.invoke_contract_cmd(
+    pc_jupyter.ml_inference_token_object.cmd_do_inference,
+    state, token_context, image=image_name, search_path=[path])
+
+# %% [markdown]
+# ### New User Performs Operations on the Asset
```

Comment:
This is pretty much a TEST not part of the factory or template. This file is supposed to be a (primitive) interface to the token, not a test file.